test/CommonSyntaxTest.h

```diff
@@ -33,13 +33,16 @@ namespace solidity::test
 struct SyntaxTestError
 {
 	std::string type;
+	std::optional<langutil::ErrorId> errorId;
 	std::string message;
 	std::string sourceName;
 	int locationStart = -1;
 	int locationEnd = -1;
 	bool operator==(SyntaxTestError const& _rhs) const
 	{
 		return type == _rhs.type &&
+			errorId.has_value() == _rhs.errorId.has_value() &&
```

Comment:
Doesn't `errorId == _rhs.errorId` do the same thing?

Reply:
Yes, after adding `ErrorId::operator==` :).